source/module_base/test/matrix3_test.cpp

```diff
@@ -16,26 +73,258 @@ class matrix3_test : public testing::Test
+	ma = get_random_matrix3();
+	mb = get_random_matrix3();
+	mc = ma + mb;
+	EXPECT_EQ(ma.e11+mb.e11, mc.e11);
+	EXPECT_EQ(ma.e12+mb.e12, mc.e12);
+	EXPECT_EQ(ma.e13+mb.e13, mc.e13);
+	EXPECT_EQ(ma.e21+mb.e21, mc.e21);
+	EXPECT_EQ(ma.e22+mb.e22, mc.e22);
+	EXPECT_EQ(ma.e23+mb.e23, mc.e23);
+	EXPECT_EQ(ma.e31+mb.e31, mc.e31);
+	EXPECT_EQ(ma.e32+mb.e32, mc.e32);
+	EXPECT_EQ(ma.e33+mb.e33, mc.e33);
+}
+
+TEST_F(Matrix3Test, MminusM)
+{
+	ModuleBase::Matrix3 ma, mb, mc;
+	ma = get_random_matrix3();
+	mb = get_random_matrix3();
+	mc = ma - mb;
+	EXPECT_EQ(ma.e11-mb.e11, mc.e11);
+	EXPECT_EQ(ma.e12-mb.e12, mc.e12);
+	EXPECT_EQ(ma.e13-mb.e13, mc.e13);
+	EXPECT_EQ(ma.e21-mb.e21, mc.e21);
+	EXPECT_EQ(ma.e22-mb.e22, mc.e22);
+	EXPECT_EQ(ma.e23-mb.e23, mc.e23);
+	EXPECT_EQ(ma.e31-mb.e31, mc.e31);
+	EXPECT_EQ(ma.e32-mb.e32, mc.e32);
+	EXPECT_EQ(ma.e33-mb.e33, mc.e33);
+}
+
+TEST_F(Matrix3Test, MoverNum)
+{
+	ModuleBase::Matrix3 ma, mb;
+	ma = get_random_matrix3();
+	mb = ma/3.0;
+	EXPECT_EQ(ma.e11/3.0,mb.e11); EXPECT_EQ(ma.e12/3.0,mb.e12); EXPECT_EQ(ma.e13/3.0,mb.e13);
+	EXPECT_EQ(ma.e21/3.0,mb.e21); EXPECT_EQ(ma.e22/3.0,mb.e22); EXPECT_EQ(ma.e23/3.0,mb.e23);
+	EXPECT_EQ(ma.e31/3.0,mb.e31); EXPECT_EQ(ma.e32/3.0,mb.e32); EXPECT_EQ(ma.e33/3.0,mb.e33);
+}
+
+TEST_F(Matrix3Test, MmultiplyM)
+{
+	ModuleBase::Matrix3 ma, mb, mc;
+	ma = get_random_matrix3();
+	mb = get_random_matrix3();
+	mc = ma * mb;
+	EXPECT_EQ( (ma.e11*mb.e11 + ma.e12*mb.e21 + ma.e13*mb.e31), mc.e11);
+	EXPECT_EQ( (ma.e11*mb.e12 + ma.e12*mb.e22 + ma.e13*mb.e32), mc.e12);
+	EXPECT_EQ( (ma.e11*mb.e13 + ma.e12*mb.e23 + ma.e13*mb.e33), mc.e13);
+	EXPECT_EQ( (ma.e21*mb.e11 + ma.e22*mb.e21 + ma.e23*mb.e31), mc.e21);
+	EXPECT_EQ( (ma.e21*mb.e12 + ma.e22*mb.e22 + ma.e23*mb.e32), mc.e22);
+	EXPECT_EQ( (ma.e21*mb.e13 + ma.e22*mb.e23 + ma.e23*mb.e33), mc.e23);
+	EXPECT_EQ( (ma.e31*mb.e11 + ma.e32*mb.e21 + ma.e33*mb.e31), mc.e31);
+	EXPECT_EQ( (ma.e31*mb.e12 + ma.e32*mb.e22 + ma.e33*mb.e32), mc.e32);
+	EXPECT_EQ( (ma.e31*mb.e13 + ma.e32*mb.e23 + ma.e33*mb.e33), mc.e33);
+}
+
+TEST_F(Matrix3Test, MmultiplyNum)
+{
+	ModuleBase::Matrix3 ma, mb, mc;
+	ma = get_random_matrix3();
+	mb = ma;
+	mc = ma;
+	mb = ma*3.0;
+	mc = 3.0*ma;
+	EXPECT_EQ(ma.e11*3.0,mb.e11); EXPECT_EQ(ma.e12*3.0,mb.e12); EXPECT_EQ(ma.e13*3.0,mb.e13);
+	EXPECT_EQ(ma.e21*3.0,mb.e21); EXPECT_EQ(ma.e22*3.0,mb.e22); EXPECT_EQ(ma.e23*3.0,mb.e23);
+	EXPECT_EQ(ma.e31*3.0,mb.e31); EXPECT_EQ(ma.e32*3.0,mb.e32); EXPECT_EQ(ma.e33*3.0,mb.e33);
+	EXPECT_EQ(ma.e11*3.0,mc.e11); EXPECT_EQ(ma.e12*3.0,mc.e12); EXPECT_EQ(ma.e13*3.0,mc.e13);
+	EXPECT_EQ(ma.e21*3.0,mc.e21); EXPECT_EQ(ma.e22*3.0,mc.e22); EXPECT_EQ(ma.e23*3.0,mc.e23);
+	EXPECT_EQ(ma.e31*3.0,mc.e31); EXPECT_EQ(ma.e32*3.0,mc.e32); EXPECT_EQ(ma.e33*3.0,mc.e33);
+}
+
+TEST_F(Matrix3Test, MmultiplyV)
+{
+	ModuleBase::Matrix3 ma;
+	ModuleBase::Vector3<double> u(3.0,4.0,5.0);
+	ModuleBase::Vector3<double> v;
+	ma = get_random_matrix3();
+	v = ma * u;
+	EXPECT_THAT(v.x, u.x*ma.e11+u.y*ma.e12+u.z*ma.e13);
```

Comment:
`EXPECT_THAT` may not be used here. Do you mean `EXPECT_EQ`?

Reply:
It's a typo. I'm surprised it passed!